rasterio/warp.py

```diff
@@ -276,9 +276,9 @@ def reproject(source, destination, src_transform=None, gcps=None,
     # If working with identity transform, assume it is crs-less data
     # and that translating the matrix very slightly will avoid #674
     eps = 1e-100
-    if src_transform and guard_transform(src_transform).is_identity:
+    if src_transform and guard_transform(np.abs(src_transform)[:6]).is_identity and src_transform.a == 1:
```

Comment:
A bit more verbose, but perhaps a bit more clear:
```
if src_transform:
    src_transform = guard_transform(src_transform)
    # if src_transform is like `identity` with positive or negative `e`,
    # translate matrix very slightly to avoid #674 and #1272.
    if src_transform.scale(1, 1).is_identity:
        src_transform = src_transform.translation(eps, eps)
```

applying `.scale(1, 1)` coerces both positive and negative `e` to be positive and thus comparable to `identity`.

Reply:
`src_transform.scale(1, 1)` returns an identity affine regardless the value of `src_transform`, so it woun't simplify the `if` statement.
`affine.scale` code:  https://github.com/sgillies/affine/blob/131854eb3e7eae3b8707ee910838ecc4bfe5ee98/affine/__init__.py#L191-L209

Reply:
Good catch, sorry about the bad advice there!  I realize I'm making this more difficult for you than necessary, thanks for the patience with me.  I'm just fishing for the conditional that will make the most sense as to why we are doing this test then adjustment in the first place (so that when we look back in a few years, it makes sense).

One last attempt:

```
from affine import identity

if src_transform:
    src_transform = guard_transform(src_transform)
    # if src_transform is like `identity` with positive or negative `e`,
    # translate matrix very slightly to avoid #674 and #1272.
    if src_transform.almost_equals(identity) or src_transform.almost_equals(Affine(1, 0, 0, 0, -1, 0)):
...
```

Does this make the intent any more clear?

Reply:
this is good, clear and explisit